File: sketch/sketch.py
#!/usr/bin/env python
import re

arr = [1, 4, 8, 9, 7, 5]

def e_arr():
    for i, v in enumerate(arr):
        yield (i, v)

for i, v in e_arr():
    print(i, v)

match = re.search('(?P<name>.*)\s+(?P<phone>.*)', 'paul 12345')
name = match.group('name')
phone = match.group('phone')
print(name, phone)

m = re.findall('H', 'hHhHH')
print(len(m))

File: sketch/sketch.rb
#!/usr/bin/env ruby

def starts_with?(string, start)
  if string =~ /^#{start}/
    return true
  else
    return false
  end
end

did_start = starts_with?('blah', 'bl')
arr = 'howdy'.chars
arr = 'howdy'.scan(/\w/)

m = 'the quick brown fox'.match(/(?<fooo>\w+)/)
puts m.is_a? Array
puts m.class
puts m.names
puts m.captures

m = "paul 12345".match(/(?<name>.*)\s+(?<phone>.*)/)
puts [m: m[:name]]

